alica_tests/src/conditions/conditions.cpp

```diff
@@ -222,4 +222,34 @@ bool TestHasNoError(const Blackboard* input, const RunningPlan* rp, const Blackb
     LockedBlackboardRO globalBlackboard(*gb);
     return !globalBlackboard.get<std::optional<std::string>>("testError").has_value();
 }
+bool ExecOrderVectorSizeCheck(const Blackboard* input, const RunningPlan* rp, const Blackboard* gb)
+{
+    LockedBlackboardRO globalBlackboard(*gb);
+    LockedBlackboardRO localBlackboard(*input);
+    if (!globalBlackboard.hasValue("execOrder")) {
+        Logging::logError("execOrder not created yet in condition...");
+        return false;
+    }
+    std::vector<std::string> execOrder = globalBlackboard.get<std::vector<std::string>>("execOrder");
+    Logging::logWarn("") << execOrder.size() << "/" << localBlackboard.get<int64_t>("expected");
+    return execOrder.size() == localBlackboard.get<int64_t>("expected");
```

Comment:
I think `expected` can be 4 here based on whether `BehAAA`'s run method is called or not.
I guess we don't need to append to execOrder in `BehAAA`'s run method but just do what is done in `PlanA`'s run method